flocker/control/_diffing.py

```diff
@@ -111,9 +262,23 @@ class Diff(PClass):
     changes = pvector_field(object)
 
     def apply(self, obj):
+        proxy = _EvolverProxy(original=obj)
         for c in self.changes:
-            obj = c.apply(obj)
-        return obj
+            if len(c.path) > 0:
+                proxy = c.apply(proxy)
+            else:
+                assert type(c) is _Set
+                proxy = _EvolverProxy(original=c.value)
+        try:
+            return proxy.commit()
+        except:
+            # Imported here to avoid circular dependencies.
+            from ._persistence import wire_encode
```

Comment:
* _diffing import wire_encode from 
* _persistence imports SERIALIZABLE_CLASSES from 
* _model imports DIFF_SERIALIZABLE_CLASSES from
* _diffing

I'll try and sort that out.